### PR TITLE
Project Auth - Caroline Söderström & Petra Eriksson

### PR DESCRIPTION
**Pull request by petrasson**
Frontend: https://auth-login-project.netlify.app/login

### CODE REVIEW

**Comment by petrasson** [comment]
Hello Annika!
Nice job merging the frontend and backend! 💪🏼 
It looks smashing and it seems to work fine. The only thing that you might want to look at is the functionality of the likes. They don't update until you refresh the page. I tried to debug it with no success. The server code looks pretty much the same as mine. So perhaps it's something in the frontend? Did it work before you added your own API? Please let me know if you find the error. I would like to understand it.

Apart from that it looks and works great! :)

File: frontend/src/Components/Login.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      .then((data) => {
+        if (data.success) {
+          batch(() => {
+            dispatch(user.actions.setUsername(data.response.username));
+            dispatch(user.actions.setUserId(data.response.id));
+            dispatch(user.actions.setAccessToken(data.response.accessToken));
+            dispatch(user.actions.setError(null));
+          });
+        } else {
+          alert("wrong credentials");
```
Comment (JosefineEngdahl) [inline frontend/src/Components/Login.js:73]: Nice solution with the alert-sign! We didn't know how to display the error message, but we really like this solution!

File: frontend/src/Components/Main.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                    dispatch(thoughts.actions.setItems(data.response));
+                    dispatch(thoughts.actions.setError(null));
+                } else {
+                    dispatch(thoughts.actions.setItems([]));
+                    dispatch(thoughts.actions.setError(data.response));
+                }
+            })
+    }, []);
+
+    const logOutOnClick = () => {
```
Comment (JosefineEngdahl) [inline frontend/src/Components/Main.js:41]: Nice with a sign out-button!

File: frontend/src/Components/Main.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        dispatch(user.actions.setAccessToken(null));
+        navigate("/login");
+    };
+
+    return (
+        <>
+            <Link to="/login">GO TO LOGIN</Link>
```
Comment (JosefineEngdahl) [inline frontend/src/Components/Main.js:48]: Do you use this button? Or can you remove it?

**Review by JosefineEngdahl — COMMENTED** [review]
Hi Caroline & Petra,

Nice job! All the requirements are met and we like that you added an "alert-sign" and sign out-button. ⭐️

We got feedback from our code reviewers that it's not working with the redirecting of "/login", and we believe it's the same for you (try enter your URL + /login and you get an error-message).  Here's an explanation of how to solve it :) --> https://stackoverflow.com/questions/55990467/catch-all-redirect-for-create-react-app-in-netlify/55990523#55990523

Great job!

/Kristin & Josefine

**Pull request closed without merging**